### DISCUSSION TITLE
Make react module federation remote targets configurable from @nrwl/react:module-federation-dev-server

### DISCUSSION THREAD

**Discussion by mathis-m** [nrwl/nx#23341] [Feature Requests]
- [x] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description

It is currently not possible to configure the target that is executed when executing @nrwl/react:module-federation-dev-server.
- It is hard coded for devRemotes here: https://github.com/nrwl/nx/blob/8762c38bb9f1bfcd66987baca486b0a6cdb9fa45/packages/react/src/executors/module-federation-dev-server/module-federation-dev-server.impl.ts#L120-L135
- And for static build here: https://github.com/nrwl/nx/blob/8762c38bb9f1bfcd66987baca486b0a6cdb9fa45/packages/react/src/executors/module-federation-dev-server/module-federation-dev-server.impl.ts#L169-L177

## Motivation

In my case the module federation is just used to compose a documentation page. Other from that the individual remotes are standalone.
Meaning the serve script and build script are primary used for the standalone serve / build.

So in general, if module federation is just used for a secondary purpose, it does not make sense to enforce the targets to be named serve and build. I might want to use these target names for my primary use case.

## Suggested Implementation

Make it configurable via options.

## Alternate Implementations

Use a name that does not conflict with default serve and build target.

**Comment by mathis-m** [comment]
If that is considered a valid feature request, I would like to know what the preferred way of configuration would be, so that I can proceed with creating a PR.

Options I see:

Global setting: this would include three new configuration options in the host to specify build, serve and serve static overrides. These would apply to all remotes.

Individual settings: this would either be a dictionary or array of overrides per remote, so that for each remote I can decide, what target to execute.

Ofc, the individual setting would be preferred, as I see the use case and a new issue about global configuration popping up.

Please join the discussion on this and state how the preferred configuration scheme would look like! Thanks.

**Comment by FrozenPandaz** [comment]
I've converted this to a discussion so we can discuss more easily.
